### PR TITLE
Class decorator

### PR DESCRIPTION
**Pull request by Gleennkar**
## Created a class decorator for OOP School Library
- Person class inherits from Nameable
- Prepare base Decorator
- Prepare CapitalizeDecorator and TrimmerDecorator
- Update Readme.md
- Check for linter errors

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
```
Comment (Ismailco) [inline base_decorator.rb]: - [ ] You have the wrong name for nameable here, the file name is `name_able.rb` kindly make sure to use the same name, you can either change the file name or change the `require_relative 'nameable'` to the same name of the file

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative 'nameable'
```
Comment (Ismailco) [inline person.rb]: - [ ] You have the wrong name for nameable here, the file name is `name_able.rb` kindly make sure to use the same name, you can either change the file name or change the `require_relative 'nameable'` to the same name of the file

File: base_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+class BaseDecorator < Nameable
```
Comment (Ismailco) [inline base_decorator.rb:3]: - [ ] Kindly make sure the nameable variable is accessible, you can do this by adding `attr_accessor :nameable` to your class

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Gleennkar, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**